SG14_test/inplace_function_test.cpp

```diff
@@ -429,6 +429,48 @@ static void RvalueRefParameter()
     g(std::make_unique<int>(42));
 }
 
+struct NoDefaultCtor
+{
+    int val;
+    explicit NoDefaultCtor(int v) : val{v} {}
+};
+
+int overloaded_function(stdext::inplace_function<int()>&& exec)
+{
+    return exec();
+}
+
+int overloaded_function(stdext::inplace_function<int(int)>&& exec)
+{
+    return exec(int{42});
+}
+
+int overloaded_function(stdext::inplace_function<int(int, NoDefaultCtor)>&& exec)
+{
+    return exec(int{42}, NoDefaultCtor{5});
+}
+
+int overloaded_function(stdext::inplace_function<void(int&)>&& exec)
+{
+    int out_parameter{88};
+    exec(out_parameter);
+    return out_parameter;
+}
+
+int fobau(NoDefaultCtor) { return 2; }
+
+
+void test_overloading()
+{
+    using brokenType = decltype(fobau(std::declval<NoDefaultCtor>()));
+    EXPECT_EQ(brokenType{2}, 2);
```

Comment:
I don't understand the point of this test case at all. `fobau` isn't overloaded, so it's super obvious that `decltype(fobau(std::declval<NoDefaultCtor>()))` is `int`, and of course `int{2} == 2`. I suspect testing/scaffolding code that was meant to be removed.

Reply:
My bad, that's leftover code, I'll remove it.